### PR TITLE
feat: search clubs and events based on location and places

### PR DESCRIPTION
**Pull request by suzy-g38**
feat:  added search feature based on your current location and states. You can either search for events near you or can check which state is having how many events.

### Related Issue 

Closes: #710

#### Screenshots  📸

https://github.com/tamalCodes/Milan/assets/62666287/9355e2ae-d42e-4997-90fc-5eacc1bb6357

### CODE REVIEW

File: src/service/MilanApi.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ export const UpdateUser = async (credentials) => {
 export const LoginUser = async (credentials) => {
   try {
     const User = await Axios.post(User_Log, credentials);
+    console.log(User);
```
Comment (tamalCodes) [inline src/service/MilanApi.js:44]: Please remove this `console.log`.

File: src/pages/display/StatesData.js
Comment (tamalCodes) [inline src/pages/display/StatesData.js:1]: Also why a lot of the states are commented ?

File: src/pages/display/ClubsPage.jsx [hunk 2]
```diff
@@ -37,15 +89,66 @@ const ClubsPage = () => {
...
             </div>
           </div>
+          <div
+            style={{
+              display: "flex",
+              justifyContent: "center",
+              alignItems: "center",
+            }}
+          >
+            <button
```
Comment (tamalCodes) [inline src/pages/display/ClubsPage.jsx:99]: If possible also remove these inline styles and add this to a separate `.css` file.

**Review by tamalCodes — CHANGES_REQUESTED** [review]
Please take a look at these changes. I will also check out your fork and run these locally to check it out for myself. Also, I think we can work on the styling. 

For example, we can improve the styles by using our `<Button/>` component, so that the styles of the button stays consistent across the page.

**Comment by suzy-g38** [comment]
@tamalCodes all the changes that you asked for, I did them. Sorry for taking so long, everytime I made a push I found a new error, it took some real time to complete every detail. Hope, the changes you asked for are made in the right way and you like them. Thanks.

File: src/components/SingleClubEvent.jsx [hunk 3]
```diff
@@ -1,11 +1,26 @@
...
+        let placeInfo = JSON.parse(event.Eventlocation).data.properties;
+        let formattedName =
+          placeInfo.name + ", " + placeInfo.city + ", " + placeInfo.state;
+        setLocation(formattedName);
+      }
+    } catch {
+      setLocation("");
+    }
+  }, []);
+
```
[the remark this answers is not in the document]
Comment (tamalCodes) [reply src/components/SingleClubEvent.jsx:27]: Hey @suzy-g38 , did you make these changes ? 
As these aren't marked `outdated` so I am just getting sure.

**Comment by tamalCodes** [comment]
Hey @suzy-g38 , i see that you made a couple of changes I suggested, again a couple of changes that I suggested had not been done. So please always click `Resolve Conversation` for the changes that you made, and for the ones you did not we can obviously talk.

For example for these changes, I suggested something which you did not do, yet the file says `outdated` as changes were made to it. So to avoid such convo - please click `Resolve Conversation` for the changes that you made.
image

If you don't it'll be tough to keep track of and I might then need to close the PR.

**Comment by suzy-g38** [comment]
> Hey @suzy-g38 , i see that you made a couple of changes I suggested, again a couple of changes that I suggested had not been done. So please always click `Resolve Conversation` for the changes that you made, and for the ones you did not, we can obviously talk.
> 
> For example for these changes, I suggested something which you did not do, yet the file says `outdated` as changes were made to it. So to avoid such convo - please click `Resolve Conversation` for the changes that you made.
> 
> image
> 
> If you don't it'll be tough to keep track of and I might then need to close the PR.

Sorry I didn't know about this resolve conversation feature. I will keep that in mind for sure. And yes there are I guess two changes from event and club page where you suggested the changes and I missed it honestly.

**Comment by tamalCodes** [comment]
No worries, no worries - just make sure you hit the button when you resolve a particular thing, given the size of this amazing feature.  Conficts and stuffs are bound to come and go, we can solve those later on.

File: src/pages/display/ClubsPage.jsx [hunk 4]
```diff
@@ -1,17 +1,69 @@
...
+    }
+    if (type === "location") {
+      if (chosenFilter === "location") {
+        setChosenFilter("");
+      } else {
+        getPositionFilter();
+        setChosenFilter("location");
+      }
+    }
+  };
```
[the remark this answers is not in the document]
Comment (suzy-g38) [reply src/pages/display/ClubsPage.jsx:59]: I could not make the change you suggested as the above code is not updating the state properly if somebody does not share their location. I also tried other ways( keeping the style you suggested) but I could not fix it. Also as I am not very used to with the suggested style of code, though I tried in many ways, there is still a possibility that maybe you can find it and be able to maintain the pattern and if you find please let me know about that I will be happy learn that.

File: src/pages/display/ClubsPage.jsx
Comment (tamalCodes) [inline src/pages/display/ClubsPage.jsx:1]: image

I think the buttons we have, consists of different inline styles, which i don't ythink will be needed. Plus they are looking totally different from our theme. So please do fix this.

Comment (tamalCodes) [reply src/pages/display/ClubsPage.jsx:1]: image

Also see how's there's an uneven spacing below ?

**Review by tamalCodes — CHANGES_REQUESTED** [review]

File: src/pages/display/ClubsPage.jsx
Comment (suzy-g38) [reply src/pages/display/ClubsPage.jsx:1]: > image
> 
> I think the buttons we have, consists of different inline styles, which i don't ythink will be needed. Plus they are looking totally different from our theme. So please do fix this.

Actually, I tried to use that first, the buttons which are already present but they have a hover effect which is creating a problem when you want to select and then deselect the button. I will then give it another try and will let you know.

Comment (suzy-g38) [reply src/pages/display/ClubsPage.jsx:1]: @tamalCodes I have made the changes that you asked for...and removed the hover. Hope you like it.

Comment (tamalCodes) [reply src/pages/display/ClubsPage.jsx:1]: image


and if for an instance there are no results, do show a message that there was nothing found.

Comment (suzy-g38) [reply src/pages/display/ClubsPage.jsx:1]: > Hi about the button part i think it needs some more love. The font family, the size it's all different. Take a look at the screenshot. Doesn't these 2 buttons looks out of place ?
> 
> Please do fix these.
> 
> image

actually, I tried that first but the buttons looked soo small, So I changed that and make it look like the home page buttons(the buttons at the centre of the home page, you will find these pair similar to the home page's button pair). And Sorry actually that the design you want I keep failing to deliver that...I am not the frontend person actually, trying this designing stuff in open source but my failure is well spotted ig. Anyway, I will try my best to make it look the way you want. Thanks.

Comment (suzy-g38) [reply src/pages/display/ClubsPage.jsx:1]: @tamalCodes I have made the changes. hope you like it.

image

**Review by tamalCodes — APPROVED** [review]
### Amazing work !
**The changes look good to me and will be merged soon.**


Do follow [Tamal](https://github.com/tamalCodes) for more Opensource fun projects and don't forget to drop a star so that you get updated about our latest [releases](https://github.com/tamalCodes/Milan/releases) (we will tag you and mention your work) and also a [shoutout](https://www.linkedin.com/feed/update/urn:li:activity:7073574804519354368/) on social media (LinkedIn and Twitter) !

Happy Opensource 🚀.

**Comment by suzy-g38** [comment]
@tamalCodes labels?

**Comment by suzy-g38** [comment]
@tamalCodes labels for gssoc?

**Pull request merged**